plotutil/xyer.go

```diff
@@ -0,0 +1,77 @@
+package plotutil
+
+import "github.com/gonum/plot/plotter"
+
+type xyer struct {
+	x, y []float64
+	len  int
+}
+
+// Len returns the length of the XYer object.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) Len() int {
+	return xy.len
+}
+
+// XY returns the x,y values at index idx.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) XY(idx int) (x, y float64) {
+	return xy.x[idx], xy.y[idx]
+}
+
+// ZipXY is a convenience function to build a plotter.XYer from slices.
+// Length will be set to the shortest slice length.
+// Panics on empty or nil slices.
+func ZipXY(x, y []float64) plotter.XYer {
```

Comment:
Move this to just below the type definition (this reflect the godoc rendering) - also below.

I'm not a fan of the function name; an idiomatic name would be `XYerFrom()` or similar.